Board.java

```diff
@@ -30,33 +28,123 @@ public Board playerMakesMove(char player, int position) {
         return this;
     }
 
+    public String getWinner() {
+        for (String combination : allWinningCombinations()) {
+            if (!combination.contains("x") && !combination.contains("-")) {
```

Comment:
could you convert this conditional into a method with a nice informative name?  I'd find it difficult actually, which is why the logic is a bit hard to understand.. Couldn't you use the method: containsOnlySame(Mark, combination/line) that you have defined below?